crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -363,6 +368,49 @@ impl BaseClient {
     }
 }
 
+/// Find the most recent decrypted event and cache it in the supplied RoomInfo.
+/// If any encrypted events are found after that one, store them in the RoomInfo
+/// too so we can use them when we get the relevant keys.
+fn cache_latest_events(room_info: &mut RoomInfo, events: &[SyncTimelineEvent]) {
+    let mut encrypted_events = Vec::with_capacity(room_info.latest_encrypted_events.capacity());
+    for e in events.iter().rev() {
```

Comment:
This iterates through timeline events again, but `handle_timeline()` already does so, and is called for both the sliding sync response and the old sync response.

Reply:
@poljar is this something I need to address or is it OK as it is now?

Reply:
It's ok for now.